File: PHP/Easy/Filters/Array/2873__Maximum_Value_of_an_Ordered_Triplet_I.php
<?php
 // https://leetcode.com/problems/maximum-value-of-an-ordered-triplet-i/
function maximumTripletValue($nums) {
    $n = count($nums);
    $maxValue = 0;

    for ($i = 1; $i < $n - 1; $i++) {
        $maxLeft = max(array_slice($nums, 0, $i));
        $maxRight = max(array_slice($nums, $i + 1));

        $value = ($maxLeft - $nums[$i]) * $maxRight;
        $maxValue = max($maxValue, $value);
    }
    return $maxValue > 0 ? $maxValue : 0;
}

File: Python/Easy/Filter/Array/2873__Maximum_Value_of_an_Ordered_Triplet_I.py
def maximumTripletValue(nums: list[int]) -> int:
    n = len(nums)
    max_value = 0

    for j in range(1, n - 1):
        max_left = max(nums[:j])
        max_right = max(nums[j + 1:])

        value = (max_left - nums[j]) * max_right
        max_value = max(max_value, value)

    return max_value if max_value > 0 else 0


nums = [12, 6, 1, 2, 7]
print(maximumTripletValue(nums))
